### PR TITLE
Link to certificate C-bindings

### PR DESCRIPTION
**Pull request by AmarOk1412**

### CODE REVIEW

File: rust/src/crypto.rs [hunk 1]
```diff
@@ -111,4 +114,58 @@ impl Drop for PrivateKey {
...
+impl Drop for DhtCertificate {
+    fn drop(&mut self) {
+        unsafe {
+            dht_certificate_delete(&mut *self)
+        }
+    }
+}
+
+impl DhtIdentity {
+    pub fn generate(common_name: &str, ca: Box<DhtIdentity>) -> DhtIdentity {
```
Comment (aberaud) [inline rust/src/crypto.rs]: `ca` can be null

File: rust/src/crypto.rs [hunk 2]
```diff
@@ -111,4 +114,58 @@ impl Drop for PrivateKey {
             dht_privatekey_delete(&mut *self)
         }
     }
+}
+
+impl DhtCertificate {
+    pub fn import(file: &str) -> io::Result<Box<DhtCertificate>> {
```
Comment (aberaud) [inline rust/src/crypto.rs]: would be nice to be able to load from memory

File: rust/src/crypto.rs (hunk 1 above)
Comment (AmarOk1412) [reply rust/src/crypto.rs]: Done

File: rust/src/crypto.rs (hunk 2 above)
Comment (AmarOk1412) [reply rust/src/crypto.rs]: Done

File: rust/src/crypto.rs [hunk 3]
```diff
@@ -111,4 +115,68 @@ impl Drop for PrivateKey {
...
+        let mut f = File::open(file)?;
+        let mut buffer = Vec::new();
+        f.read_to_end(&mut buffer)?;
+        Ok(DhtCertificate::from_bytes(&buffer))
+    }
+
+    pub fn from_bytes(buffer: &Vec<u8>) -> Box<DhtCertificate> {
```
Comment (aberaud) [inline rust/src/crypto.rs]: allow to import from slice ?

File: rust/src/infohash.rs [hunk 4]
```diff
@@ -47,6 +47,14 @@ impl InfoHash {
         h
     }
 
+    pub fn from_bytes(data: &Vec<u8>) -> InfoHash {
```
Comment (aberaud) [inline rust/src/infohash.rs]: allow to import from slice ?

**Pull request merged**